crates/gpui/src/platform/windows/window.rs

```diff
@@ -109,75 +140,413 @@ impl WindowsWindowInner {
+            } else {
+                CallbackResult::Handled(false)
             }
-            WM_DESTROY => {
-                let mut callbacks: std::cell::RefMut<'_, Callbacks> = self.callbacks.borrow_mut();
-                if let Some(callback) = callbacks.close.take() {
-                    callback()
-                }
-                let mut window_handles = self.platform_inner.window_handles.borrow_mut();
-                window_handles.remove(&self.handle);
-                if window_handles.is_empty() {
-                    self.platform_inner
-                        .foreground_executor
-                        .spawn(async {
-                            unsafe { PostQuitMessage(0) };
-                        })
-                        .detach();
+        } else {
+            CallbackResult::Unhandled
+        }
+    }
+
+    fn handle_keyup_msg(&self, wparam: WPARAM) -> CallbackResult {
+        let mut callbacks: std::cell::RefMut<'_, Callbacks> = self.callbacks.borrow_mut();
+        let keystroke = self.parse_key_msg_keystroke(wparam);
+        if let Some(keystroke) = keystroke {
+            if let Some(callback) = callbacks.input.as_mut() {
+                let event = KeyUpEvent { keystroke };
+                CallbackResult::Handled(callback(PlatformInput::KeyUp(event)))
+            } else {
+                CallbackResult::Handled(false)
+            }
+        } else {
+            CallbackResult::Unhandled
+        }
+    }
+
+    fn handle_char_msg(&self, wparam: WPARAM) -> LRESULT {
+        let mut callbacks: std::cell::RefMut<'_, Callbacks> = self.callbacks.borrow_mut();
+        if let Some(callback) = callbacks.input.as_mut() {
+            let modifiers = self.current_modifiers();
+            let msg_char = wparam.0 as u8 as char;
+            let keystroke = Keystroke {
+                modifiers,
+                key: msg_char.to_string(),
+                ime_key: Some(msg_char.to_string()),
+            };
+            let event = KeyDownEvent {
+                keystroke,
+                is_held: false,
+            };
+
+            if callback(PlatformInput::KeyDown(event.clone())) {
+                return LRESULT(0);
+            }
+
+            if let Some(mut input_handler) = self.input_handler.take() {
+                if let Some(ime_key) = &event.keystroke.ime_key {
+                    input_handler.replace_text_in_range(None, ime_key);
                 }
-                return LRESULT(1);
+                self.input_handler.set(Some(input_handler));
+                return LRESULT(0);
             }
-            _ => return unsafe { DefWindowProcW(self.hwnd, msg, wparam, lparam) },
         }
-        LRESULT(0)
+        return LRESULT(1);
+    }
+
+    fn handle_mouse_down_msg(&self, button: MouseButton, lparam: LPARAM) -> LRESULT {
+        let mut callbacks: std::cell::RefMut<'_, Callbacks> = self.callbacks.borrow_mut();
```

Comment:
Curious why we need to specify the type. We don't do this in Linux platform.

Reply:
not necessary. was a copy-paste from https://github.com/zed-industries/zed/pull/8490/files#diff-b38258c3b6a51a8f791374836458afd4612ac04d28b558e53b9de1e073488282R162

will update.